packages/react/src/components/Checkbox/Checkbox.test.tsx

```diff
@@ -46,6 +46,29 @@ test('should render disabled checked checkbox', () => {
   expect(input).toBeChecked();
 });
 
+test('should render indeterminate checkbox', () => {
+  const ref = createRef<HTMLInputElement>();
+  render(<Checkbox id="id" label="checkbox" ref={ref} indeterminate />);
+  expect(ref.current).toBeInstanceOf(HTMLInputElement);
+  expect(ref.current?.indeterminate).toBeTruthy();
+  expect(ref.current).toEqual(
+    screen.queryByRole('checkbox', { name: 'checkbox' })
+  );
```

Comment:
This isn't quite how we should be doing the assertions here. We should be limiting to just what should be tested for this specific test, i.e. that an indeterminate checkbox was rendered. The way this test is written we're testing that the ref value has an indeterminate value but not that a indeterminate value was rendered on the page. Please take a look at how some of the other tests are written here to try to eliminate down to only the assertions that are necessary.

Additionally, jest-dom already has a matcher we can use, [`toBePartiallyChecked()`](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tobepartiallychecked)

Reply:
Thanks, updated tests